app/models/user.rb

```diff
@@ -83,6 +83,9 @@ class User < ApplicationRecord
   validates :dribbble_url,
               allow_blank: true,
               format: /\Ahttps:\/\/(www.dribbble.com|dribbble.com)\/([a-zA-Z0-9\-\_]{2,20})\/?\Z/
+  validates :medium_url,
+              allow_blank: true,
+              format: /\Ahttps:\/\/(www.medium.com|medium.com)\/([a-zA-Z0-9\-\_\@]{2,20})\/?\Z/
```

Comment:
Looks like the max length for Medium usernames is 30

Reply:
Also they only allow periods and underscores (and the @ symbol).

Reply:
@Zhao-Andy Can you provide a reference for the medium username rules? eg link to webpage where this is defined?

Reply:
When I manually tried to change my username it allowed for 32 chars including an @ symbol at the start, which is different from your claim the max length is 30

Reply:
Agreed on needing to add the . and _ to the valid chars list

Reply:
I've now added periods to the valid chars in the medium url. The underscore and @ symbol are in there already.

Reply:
Added a commit to extend the max length to 32 chars 453b3e1

Reply:
@Zhao-Andy The reason I left in the hyphen is because if an individual wants to link to a publication, for example I publish some of my posts in my own publication https://medium.com/the-code-review, then the medium validation would not permit this.

I guess this depends on how strict we want to be about only allowing usernames to be included, or whether we want to allow people to link to publications on medium as well in the medium url

Reply:
I think linking to publications is valid, so allowing dashes makes sense. I got the max username length from their HTML validation. A max of 32 characters sounds good to me.